c10/core/DefaultDtype.cpp

```diff
@@ -3,12 +3,21 @@
 
 namespace c10 {
 static auto default_dtype = caffe2::TypeMeta::Make<float>();
+static auto default_complex_dtype = caffe2::TypeMeta::Make<std::complex<float>>();
 
 void set_default_dtype(caffe2::TypeMeta dtype) {
   default_dtype = std::move(dtype);
+  if(dtype == caffe2::TypeMeta::Make<float>()) {
```

Comment:
err...what happens if you set a complex type to be the default dtype?  Is that allowed?

Reply:
you can't set a complex type to be default. the default for complex is inferred based on the current default(float/double --> ComplexFloat/ComplexDouble)

Reply:
When you set half as the default scalar type (admittedly a bad idea) won't this make the corresponding complex type double and not float? Maybe the mapping should be double->double and everything else to float for safety?

Reply:
good idea! will do